compiler/src/dotty/tools/dotc/Run.scala

```diff
@@ -68,6 +69,12 @@ class Run(comp: Compiler, ictx: Context) extends ImplicitRunInfo with Constraint
   private[this] var myUnitsCached: List[CompilationUnit] = _
   private[this] var myFiles: Set[AbstractFile] = _
 
+  /** Units that are added because of source completers but that are not
+   *  compiled in current run.
+   */
+  val lateUnits = mutable.ListBuffer[CompilationUnit]()
```

Comment:
Would be nicer if the mutability was hidden from outside this class, like for units and files.